src/algorithm/mod.rs

```diff
@@ -31,9 +32,9 @@ pub enum ButterflyEnum<T> {
 	Butterfly2(butterflies::Butterfly2),
 	Butterfly3(butterflies::Butterfly3<T>),
 	Butterfly4(butterflies::Butterfly4),
-	Butterfly5(butterflies::Butterfly5<T>),
+	Butterfly5(Box<butterflies::Butterfly5<T>>),
```

Comment:
Why are `Butterfly5` and `Butterfly7` boxed now? Is this just to save space on the enum size?